src/Uno.UI.Runtime.Skia.Wpf/Islands/UnoXamlHostBase.cs

```diff
@@ -0,0 +1,361 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Uno.UI.Runtime.Skia.Wpf.Hosting;
+using Uno.UI.Skia.Platform;
+using Uno.UI.XamlHost;
+using WUX = Windows.UI.Xaml;
+
+namespace Uno.UI.XamlHost.Skia.Wpf
+{
+	/// <summary>
+	/// UnoXamlHost control hosts UWP XAML content inside the Windows Presentation Foundation
+	/// </summary>
+	abstract partial class UnoXamlHostBase
+	{
+		/// <summary>
+		/// An instance of <seealso cref="IXamlMetadataContainer"/>. Required to
+		/// probe at runtime for custom UWP XAML type information.
+		/// This must be implemented by the instance of <seealso cref="WUX.Application"/>
+		/// </summary>
+		/// <remarks>
+		/// <seealso cref="WUX.Application"/> object is required for loading custom control metadata.  If a custom
+		/// Application object is not provided by the application, the host control will create an instance of <seealso cref="XamlApplication"/>.
+		/// Instantiation of the application object must occur before creating the DesktopWindowXamlSource instance.
+		/// If no Application object is created before DesktopWindowXamlSource is created, DestkopWindowXamlSource
+		/// will create an instance of <seealso cref="XamlApplication"/> that implements <seealso cref="IXamlMetadataContainer"/>.
+		/// </remarks>
+		private static readonly IXamlMetadataContainer _metadataContainer;
+
+		static UnoXamlHostBase()
+		{
+			//TODO:MZ: This should be set in a different location, possibly in a more general way
+			if (MetadataProviderDiscovery.MetadataProviderFactory is null)
+			{
+				MetadataProviderDiscovery.MetadataProviderFactory = type =>
+				{
+					WUX.Application application = null;
+					var host = new WpfIslandsHost(
```

Comment:
Should host be stored somewhere or it it rooted by something else?

Reply:
Fixed